### DISCUSSION TITLE
TypeScript: Setting string id to reference

### DISCUSSION THREAD

**Discussion by sethdumaguin** [mobxjs/mobx-state-tree#1880] [Q&A] [answered]
Hi there! 

I have an action in my model that takes in an `id` typed `string` and sets it to safeReference in the `setNode` action. However, I am getting this ts error. I wonder if I am doing this correctly and if there's any help I can get. Would really appreciate it!

![Screen Shot 2022-03-01 at 12 42 07 PM](https://user-images.githubusercontent.com/21695860/156245616-338636d5-4a31-437c-8bb4-115debeba58e.png)

Here's is a small snippet of what the structure of the code is below.

```
const Node = types.model({
  id: types.identifier,
})

const RootStore = types.model({
  node: types.safeReference(Node)
}).actions((self) => ({
  setNode: (id: string) => {
    self.node = cast(id);
  }
})


```

**Comment by EmilTholin** [comment, marked as answer]
Hi @sethdumaguin!

At this time you have to cast it to `any` sadly, [as outlined by xaviergonz here](https://github.com/mobxjs/mobx-state-tree/issues/1123#issuecomment-491369379).

```ts
const Node = types.model({
  id: types.identifier
});

const RootStore = types
  .model({
    node: types.safeReference(Node)
  })
  .actions((self) => ({
    setNode: (id: string) => {
      self.node = id as any;
    }
  }));
```

**Reply by sethdumaguin** [reply]
Thank you so much for the answer! 🙏
